src/connector/src/sink/encoder/json.rs

```diff
@@ -18,21 +18,24 @@ use base64::engine::general_purpose;
 use base64::Engine as _;
 use chrono::{Datelike, Timelike};
 use indexmap::IndexMap;
+use itertools::Itertools;
 use risingwave_common::array::{ArrayError, ArrayResult};
 use risingwave_common::catalog::{Field, Schema};
 use risingwave_common::row::Row;
 use risingwave_common::types::{DataType, DatumRef, Decimal, ScalarRefImpl, ToText};
 use risingwave_common::util::iter_util::ZipEqDebug;
 use serde_json::{json, Map, Value};
 
-use super::{CustomJsonType, Result, RowEncoder, SerTo, TimestampHandlingMode};
+use super::{CustomJsonType, KafkaConnectParams, Result, RowEncoder, SerTo, TimestampHandlingMode};
+use crate::sink::formatter::debezium_json::field_to_json;
 use crate::sink::SinkError;
 
 pub struct JsonEncoder {
     schema: Schema,
     col_indices: Option<Vec<usize>>,
     timestamp_handling_mode: TimestampHandlingMode,
     custom_json_type: CustomJsonType,
+    kafka_connect: Option<KafkaConnectParams>,
```

Comment:
how about name it `EncodeExtParams`, not limit to kafka connect, and make it a non-option field? And we can set different defaults to each format.

Reply:
I am viewing this parameter as dedicated to kafka connect. If there was `EncodeExtParams`, it would look like:
```
pub struct JsonEncoder {
    schema: Schema,
    col_indices: Option<Vec<usize>>,
    extra_params: EncodeExtParams,
}
pub struct EncodeExtParams {
    timestamp_handling_mode: TimestampHandlingMode,
    custom_json_type: CustomJsonType,
    kafka_connect: Option<KafkaConnectParams>,
}
```
and the extra nesting seems unnecessary.

Reply:
> I am viewing this parameter as dedicated to kafka connect. 

I think there will be more and more options and we cannot always add to encoder props. An extra Params struct can apply to other encoders with different default values. 

```rust
pub struct EncodeExtParams {
    timestamp_handling_mode: TimestampHandlingMode,
    custom_json_type: CustomJsonType,
    kafka_connect: Option<KafkaConnectParams>,
}

impl EncodeExtParams {
    const DEBEZIUM = EncodeExtParams { .. }
    const UPSERT = EncodeExtParams { .. }
}
```

Reply:
> An extra Params struct can apply to other encoders with different default values.

First of all, we are discussing about `encoder` params rather than `formatter` params. So a proper example is whether json/proto/avro/csv/template would share a lot of parameters. It would be an anti-pattern to:
```
pub struct EncodeExtParams {
    // only json and proto, not avro
    timestamp_handling_mode: TimestampHandlingMode,
    // only json
    custom_json_type: CustomJsonType,
    // json, proto, and avro, but not template
    kafka_connect: Option<KafkaConnectParams>,
    // csv
    delimiter: &str,
    // only template
    template: &str,
}
```

But it is just a difference in our predictions of future. If it turns out to be more common options, I agree they should be extracted to a common struct.